registry/client/errors.go

```diff
@@ -109,15 +90,25 @@ func parseHTTPErrorResponse(statusCode int, r io.Reader) error {
 func HandleErrorResponse(resp *http.Response) error {
 	if resp.StatusCode >= 400 && resp.StatusCode < 500 {
 		// Check for OAuth errors within the `WWW-Authenticate` header first
+		// See https://tools.ietf.org/html/rfc6750#section-3
 		for _, c := range challenge.ResponseChallenges(resp) {
 			if c.Scheme == "bearer" {
-				errStr := c.Parameters["error"]
-				if errStr != "" {
-					return &OAuthError{
-						ErrorCode:   errStr,
-						Description: c.Parameters["error_description"],
-					}
+				var err errcode.Error
+				// codes defined at https://tools.ietf.org/html/rfc6750#section-3.1
+				switch c.Parameters["error"] {
+				case "invalid_token":
+					err.Code = errcode.ErrorCodeUnauthorized
```

Comment:
They potentially have different messages, from the client perspective right now we only use the first in the list. We currently don't have a defined use of the remainder, we should probably at least log them.